lib/openhab/dsl/items/builder.rb

```diff
@@ -582,9 +582,9 @@ def expire(*args, command: nil, state: nil, ignore_state_updates: nil, ignore_co
         def unit=(unit)
           @unit = unit
 
-          self.dimension ||= "Temperature" if unit&.to_s == "mired"
-          self.dimension ||= unit && org.openhab.core.types.util.UnitUtils.parse_unit(unit)&.then do |u|
-            org.openhab.core.types.util.UnitUtils.get_dimension_name(u)
+          if (openhab_unit = unit && org.openhab.core.types.util.UnitUtils.parse_unit(unit))
+            self.dimension ||= "Temperature" if openhab_unit == org.openhab.core.library.unit.Units::MIRED
```

Comment:
Using `Units::MIRED` is enough, no need for the full path. `Units` is imported by default in the default scope.